test/test_tensorboard.py

```diff
@@ -348,56 +352,57 @@ def test_video(self):
+            self.assertTrue(compare_proto(summary.video('dummy', tensor_N(shape=(4, 3, 1, 8, 8))), self))
+            summary.video('dummy', np.random.rand(16, 48, 1, 28, 28))
+            summary.video('dummy', np.random.rand(20, 7, 1, 8, 8))
 
         def test_audio(self):
-            self.assertTrue(compare_proto(summary.audio('dummy', np.random.rand(42)), self))
+            self.assertTrue(compare_proto(summary.audio('dummy', tensor_N(shape=(42,))), self))
 
         def test_text(self):
             self.assertTrue(compare_proto(summary.text('dummy', 'text 123'), self))
 
         def test_histogram_auto(self):
-            self.assertTrue(compare_proto(summary.histogram('dummy', np.random.rand(1024), bins='auto', max_bins=5), self))
+            self.assertTrue(compare_proto(summary.histogram('dummy', tensor_N(shape=(1024,)), bins='auto', max_bins=5), self))
 
         def test_histogram_fd(self):
-            self.assertTrue(compare_proto(summary.histogram('dummy', np.random.rand(1024), bins='fd', max_bins=5), self))
+            self.assertTrue(compare_proto(summary.histogram('dummy', tensor_N(shape=(1024,)), bins='fd', max_bins=5), self))
 
         def test_histogram_doane(self):
-            self.assertTrue(compare_proto(summary.histogram('dummy', np.random.rand(1024), bins='doane', max_bins=5), self))
+            self.assertTrue(compare_proto(summary.histogram('dummy', tensor_N(shape=(1024,)), bins='doane', max_bins=5), self))
+
+        def test_custom_scalars(self):
+            layout = {'Taiwan': {'twse': ['Multiline', ['twse/0050', 'twse/2330']]},
+                      'USA': {'dow': ['Margin', ['dow/aaa', 'dow/bbb', 'dow/ccc']],
+                              'nasdaq': ['Margin', ['nasdaq/aaa', 'nasdaq/bbb', 'nasdaq/ccc']]}}
+            self.assertTrue(compare_proto(summary.custom_scalars(layout), self))
 
     def remove_whitespace(string):
         return string.replace(' ', '').replace('\t', '').replace('\n', '')
 
     def compare_proto(str_to_compare, function_ptr):
-        # TODO: enable test after tensorboard is ready.
-        return True
-        if 'histogram' in function_ptr.id():
-            return  # numpy.histogram has slight difference between versions
-
-        if 'pr_curve' in function_ptr.id():
-            return  # pr_curve depends on numpy.histogram
 
         module_id = function_ptr.__class__.__module__
+        test_dir = os.path.dirname(sys.modules[module_id].__file__)
         functionName = function_ptr.id().split('.')[-1]
-        test_file = os.path.realpath(sys.modules[module_id].__file__)
-        expected_file = os.path.join(os.path.dirname(test_file),
+        expected_file = os.path.join(test_dir,
                                      "expect",
-                                     module_id.split('.')[-1] + '.' + functionName + ".expect")
+                                     'TestTensorboard.' + functionName + ".expect")
```

Comment:
Nit - can we change `TestTensorboard.*.expect` to `TestTensorBoard.*.expect` with the B capitalized?